### PR TITLE
DOCS-7436 - Serverless | Azure App Service - Node.js extension for Windows

### PR DESCRIPTION
**Pull request by cswatt**
### What does this PR do? What is the motivation?
- Adds windows node support to table on Azure App Service index page
- Adds install instructions to Azure App Service - Windows page

### Merge instructions

- [ ] Please merge after reviewing

#### questions for duncan or other reviewer
- for step 8, it just says "select the Datadog APM Node extension". is there an official name for this? will users have to search for it?
- for the dogstatsd instructions, I copied over what was in the java instructions. in step 1, the "DogStatsD client" linked to a [maven package](https://search.maven.org/artifact/com.datadoghq/java-dogstatsd-client) so for node, i have this linking to [hot-shots](https://github.com/brightcove/hot-shots)—is this correct?
- the dogstatsd instructions put installing the extension at step 4, which suggests it is mandatory to only install the extension AFTER you set up dogstatsd. is this actually the case? if not, i'll update it to "If you have not already, install the extension" for all three languages.

### CODE REVIEW

**Comment by duncanpharvey** [comment]
Addressing your questions in the description

- for step 8, it just says "select the Datadog APM Node extension". is there an official name for this? will users have to search for it?

Here's a screenshot of what users seen in Azure when adding a site extension. The exact name of the extension is `Node Datadog APM(<version>)--Datadog`.

Screenshot 2024-03-15 at 3 44 31 PM

- for the dogstatsd instructions, I copied over what was in the java instructions. in step 1, the "DogStatsD client" linked to a [maven package](https://search.maven.org/artifact/com.datadoghq/java-dogstatsd-client) so for node, i have this linking to [hot-shots](https://github.com/brightcove/hot-shots)—is this correct?

A dogstatsd client was recently added to the Node.js tracer so we are actually using the `dd-trace` library to submit custom metrics via DogStatsD. Since `dd-trace` is included as a package in the extension and is appended to the `NODE_PATH`, it does not need to be added as a dependency in the user's project.

However, since the extension can't be used for local testing we'll need to update these instructions slightly. What do you think of this? Feel free to reword as needed!

------
The Azure App Service extension includes an instance of [DogStatsD](https://docs.datadoghq.com/developers/dogstatsd) (Datadog’s metrics aggregation service). This enables you to submit custom metrics, service checks, and events directly to Datadog from Azure Web Apps with the extension.

Writing custom metrics and checks in this environment is similar to the process for doing so with an application on a standard host running the Datadog Agent. To submit custom metrics to Datadog from Azure App Service using the extension:

1. Initialize DogStatsD and write custom metrics in your application.
2. Deploy your code to a supported Azure web app.
3. If you have not already, install the Datadog App Service extension.

Note that the Node DogStatsD client is part of the Node.js tracer that is packaged in the extension and is automatically appended to the `NODE_PATH`. As such, the Node DogStatsD client does not need to be added as a dependency to your project except in the case of local testing. Reference the [release notes](https://github.com/DataDog/datadog-aas-extension/releases) to find the version of the Node.js tracer that is packaged with a specific version of the extension.

------

- the dogstatsd instructions put installing the extension at step 4, which suggests it is mandatory to only install the extension AFTER you set up dogstatsd. is this actually the case? if not, i'll update it to "If you have not already, install the extension" for all three languages.

I agree it makes sense to add `If you have not already` since there could be a scenario where users set up and used an extension with their app then add custom metrics at a later date.

**Comment by duncanpharvey** [comment]
The rest of the doc looks good 👍

File: content/en/serverless/azure_app_services/_index.md [hunk 1]
```diff
@@ -69,3 +70,4 @@ Capabilities:
 [10]: /serverless/azure_app_services/azure_app_services_linux?tab=java
 [11]: /developers/dogstatsd/
 [12]: /serverless/azure_app_services/azure_app_services_container
+[13]: /serverless/azure_app_services/azure_app_services_windows?tab=node#setup
```
Comment (brandondatadog) [inline content/en/serverless/azure_app_services/_index.md:73]: When I click on the link on the staging site it still directs to the .NET tab

File: content/en/serverless/azure_app_services/azure_app_services_windows.md [hunk 2]
```diff
@@ -244,6 +244,87 @@ Learn more about [custom metrics][10].
 [9]: /developers/dogstatsd/?tab=java#code
 [10]: /metrics/
 {{% /tab %}}
+{{% tab "Node.js" %}}
+### Requirements
+
+If you haven't already, set up the [Datadog-Azure integration][1] first.
+
+Datadog's Azure App Service Node.js extension supports Azure App Service Web Apps. Function Apps are not supported.
```
Comment (brandondatadog) [inline content/en/serverless/azure_app_services/azure_app_services_windows.md:252]:
```suggestion
Datadog's Azure App Service Node.js extension supports Azure App Service Web Apps. Function Apps are not yet supported.
```

**Review by brandondatadog — CHANGES_REQUESTED** [review]

File: content/en/serverless/azure_app_services/azure_app_services_windows.md (hunk 2 above)
Comment (cswatt) [reply content/en/serverless/azure_app_services/azure_app_services_windows.md:252]: As a matter of documentation policy, we do not use the phrasing "not yet" or anything that suggests the future existence of functionality that is not presently available to customers.

File: content/en/serverless/azure_app_services/_index.md (hunk 1 above)
Comment (kayayarai) [inline content/en/serverless/azure_app_services/_index.md]:
```suggestion
[13]: /serverless/azure_app_services/azure_app_services_windows?tab=nodejs#setup
```
I think that will fix the link

File: content/en/serverless/azure_app_services/azure_app_services_windows.md [hunk 3]
```diff
@@ -244,6 +244,87 @@ Learn more about [custom metrics][10].
...
+
+<div class="alert alert-info">Datadog's Node.js tracer, <code>dd-trace</code>, is packaged in the Azure App Services extension. It is automatically appended to the <code>NODE_PATH</code>.<br/><br/> <strong>You do not need to add</strong> <code>dd-trace</code> <strong>as a dependency in</strong> <code>package.json</code>. Explicitly adding <code>dd-trace</code> as a dependency may override the version provided by the extension. For local testing, reference the <a href="https://github.com/DataDog/datadog-aas-extension/releases">release notes</a> to find the appropriate version of the Node.js tracer for your version of the Azure App Service extension.</div>
+
+Learn more about [custom metrics][10].
+
+[1]: /integrations/azure
+[2]: /tracing/setup/java/
+[3]: https://portal.azure.com/
+[4]: https://app.datadoghq.com/organization-settings/api-keys
+[5]: /tracing/trace_collection/library_config/dotnet-framework/#configuration-settings
```
Comment (kayayarai) [inline content/en/serverless/azure_app_services/azure_app_services_windows.md:320]:
```suggestion
[5]: /tracing/trace_collection/library_config/nodejs/#configuration-settings
```

File: content/en/serverless/azure_app_services/azure_app_services_windows.md [hunk 4]
```diff
@@ -244,6 +244,87 @@ Learn more about [custom metrics][10].
...
+
+<div class="alert alert-info">Azure resource logs do not include trace ID. <br/><br/>Trace ID injection occurs inside your application. Azure resource logs are generated by Azure in the management plane, and therefore do not include the trace ID.</div>
+
+### Custom metrics with DogStatsD
+
+Datadog's Azure App Service Node.js extension includes an instance of [DogStatsD][7], Datadog's metrics aggregation service. This enables you to submit custom metrics, service checks, and events directly to Datadog from Azure Web Apps.
+
+Writing custom metrics and checks in this environment is similar to the process for doing so with an application on a standard host running the Datadog Agent. To submit custom metrics to Datadog from Azure App Service using the extension:
+
+1. Initialize DogStatsD and write custom metrics in your application.
```
Comment (kayayarai) [inline content/en/serverless/azure_app_services/azure_app_services_windows.md:296]: Maybe add a link to instructions on doing this

**Review by kayayarai — CHANGES_REQUESTED** [review]

**Review by kayayarai — APPROVED** [review]

**Pull request merged**